tests/test_units/test_dataio.py

```diff
@@ -535,12 +535,13 @@ def test_exportdata_no_iter_folder(
     edata = ExportData(config=rmsglobalconfig, content="depth")
     assert edata._fmurun is True
 
-    edata.export(regsurf)
-
-    assert edata._metadata["fmu"]["realization"]["name"] == "realization-1"
-    assert edata._metadata["fmu"]["realization"]["id"] == 1
-    assert edata._metadata["fmu"]["iteration"]["name"] == "iter-0"
-    assert edata._metadata["fmu"]["iteration"]["id"] == 0
+    out = Path(edata.export(regsurf))
+    with open(out.parent / f".{out.name}.yml", encoding="utf-8") as f:
+        metadata = yaml.safe_load(f)
```

Comment:
any particular reason you are not just ? 
```
metadata = edata.generate_metadata(obj)
```

Reply:
I thought about it changing it as such, but these tests seem written specifically for the actual export of the metadata, so I figured reading what was exported felt appropriate. I am guessing that checking the metadata stored internally was more for convenience over opening the file